Lesson15/nurtap/script.js

```diff
@@ -0,0 +1,390 @@
+/*
+4. Sum of Numbers in a Range (While Loop)
+    - Define a function `sumRange(start, end)` that uses a while loop
+      to sum all integers from `start` to `end` (inclusive).
+    - Log the final sum.
+*/
+
+function sumRange(start, end) {
+  let total = 0;
+  let j = start;
+  while (j <= end) {
+    total += j;
+    j++;
+  }
+  console.log(total);
+}
+sumRange(2, 5);
+
+/*
+5. Reverse an Array
+   - Define a function `reverseArray(arr)` that reverses the elements
+     of an array manually using a for loop (without using .reverse()).
+   - Log the reversed array.
+*/
+
+function reverseArray(arr) {
+  let reversedArr = [];
+  for (let i = 0; i < arr.length; i++) {
+    reversedArr.push(arr[arr.length - (i + 1)]);
+  }
+  console.log(reversedArr);
+}
+reverseArray(["a", "b", "c", "d"]);
+
+/*
+6. Filter Out Negative Numbers
+   - Define a function `filterNegative(numbers)` that loops through
+     an array of numbers and creates a new array without any negative values.
+   - Log the new array.
+*/
+
+function filterNegative(numbers) {
+  let positiveNumbers = [];
+  for (let i = 0; i < numbers.length; i++) {
+    if (numbers[i] >= 0) {
+      positiveNumbers.push(numbers[i]);
+    }
+  }
+  console.log(positiveNumbers);
+}
+filterNegative([0, 1, 2, -3, 4, -5]);
+
+/*
+7. Double the Values (For-of Loop)
+   - Define a function `doubleValues(numbers)` that uses a for-of loop
+     to multiply each number by 2, storing results in a new array.
+   - Log the new array.
+*/
+
+function doubleValues(numbers) {
+  let newNumbers = [];
+  for (let number of numbers) {
+    newNumbers.push(number * 2);
+  }
+  console.log(newNumbers);
+}
+
+doubleValues([1, 0, 2, 3, 5]);
+
+/*
+8. Print Each Character of a String (For-of)
+   - Define a function `printCharacters(str)` that uses a for-of loop
+     to log each character in the string on a separate line.
+*/
+
+function printCharacters(str) {
+  for (let character of str) {
+    console.log(character);
+  }
+}
+printCharacters("nurtap");
+
+/*
+9. Sum All Values in an Object
+   - Define a function `sumObjectValues(obj)` that iterates over the
+     properties of an object (using a for-in loop) and sums all numeric values.
+   - Log the sum.
+   - Example: {a: 10, b: 20, c: 5} -> 35
+*/
+
+function sumObjectValues(obj) {
+  let sum = 0;
+  for (let key in obj) {
+    sum += obj[key];
+  }
+  console.log(sum);
+}
+sumObjectValues({ a: 10, b: 20, c: 5 });
+/*
+10. Print Keys of an Object (For-in)
+    - Define a function `printObjectKeys(obj)` that uses a for-in loop
+      to log each key of the object.
+    - Example: { name: "Alice", age: 25 } -> logs "name", then "age"
+*/
+
+function printObjectKeys(obj) {
+  for (const key in obj) {
+    console.log(obj[key]);
```

Comment:
You only need to print the key, so print it directly like this:
```suggestion
    console.log(key);
```